src/frameworks/next/utils.ts

```diff
@@ -0,0 +1,63 @@
+import type { Header, Redirect, Rewrite } from "next/dist/lib/load-custom-routes";
+import { isUrl } from "../utils";
+
+/**
+ * Whether the given path has a regex or not.
+ * According to the Next.js documentation:
+ * ```md
+ *  To match a regex path you can wrap the regex in parentheses
+ *  after a parameter, for example /post/:slug(\\d{1,}) will match /post/123
+ *  but not /post/abc.
+ * ```
+ * See: https://nextjs.org/docs/api-reference/next.config.js/redirects#regex-path-matching
+ */
+export function pathHasRegex(path: string): boolean {
+  // finds parentheses that are not preceded by double backslashes
+  const regex = /(?<!\\)\(/;
+
+  if (regex.test(path)) {
+    return true;
+  }
+
+  return false;
+}
+
+/**
+ * Remove double backslashes from a string
+ */
+export function cleanEscapedChars(path: string): string {
+  return path.replace(/\\/g, "");
```

Comment:
Shouldn't this replace with a single "\", rather than empty? mind getting some more edge cases covered in the test?

Reply:
@jamesdaniels right.. that should be the case. Reviewing it now Next.js has this on their docs:

> The following characters (, ), {, }, :, *, +, ? are used for regex path matching, so when used in the source as non-special values they must be escaped by adding

I'm fixing it and adding test cases, will push

Reply:
@leoortizz `path.replace(/\\([\(\){}\:\+\?])/g, (a,b) => b)` should do it IMO. WDYT?

```
> '/\\(\\)\\{\\}\\:\\+\\?/:slug'.replace(/\\([\(\){}\:\+\?])/g, (a,b) => b)
'/(){}:+?/:slug'
```

Reply:
@jamesdaniels this seem to fix it, thanks! I implemented it and adjusted the unit tests, should be all working now